libraries/commonIO/src/main/java/org/eclipse/sw360/commonIO/AttachmentFrontendUtils.java

```diff
@@ -34,18 +34,17 @@
 public class AttachmentFrontendUtils {
 
     private static final Logger log = Logger.getLogger(AttachmentFrontendUtils.class);
-    protected final AttachmentService.Iface client;
 
     private AttachmentStreamConnector connector;
     // TODO add Config class and DI
     private final Duration downloadTimeout = Duration.durationOf(30, TimeUnit.SECONDS);
 
-    public AttachmentFrontendUtils() {
-        this(new ThriftClients());
-    }
+    protected final ThreadLocal<AttachmentService.Iface> attchmntClient = ThreadLocal.<AttachmentService.Iface>withInitial(
+            () -> {
+                return new ThriftClients().makeAttachmentClient();
+            });
 
-    public AttachmentFrontendUtils(ThriftClients thriftClients) {
-        client = thriftClients.makeAttachmentClient();
+    public AttachmentFrontendUtils() {
```

Comment:
This one might not be needed since this default constructor is the only one. But it's okay for now :-)